src/main/java/slimeknights/tconstruct/tools/modifiers/ability/interaction/BlockTransformModifier.java

```diff
@@ -127,6 +129,80 @@ public InteractionResult afterBlockUse(IToolStackView tool, int level, UseOnCont
     return didTransform ? InteractionResult.sidedSuccess(world.isClientSide) : InteractionResult.PASS;
   }
 
+  @Override
+  public boolean onDispenserUse(IToolStackView tool, int level, BlockSource source, ItemStack stack) {
+    // tool must not be broken
+    if (tool.isBroken()) {
+      return false;
+    }
+
+    UseOnContext context = ModifiableItem.contextFromBlockSource(source, stack);
+
+    // for hoes and shovels, must have nothing but plants above
+    if (requireGround && context.getClickedFace() == Direction.DOWN) {
+      return true;
+    }
+
+    // must actually transform
+    Level world = context.getLevel();
+    BlockPos pos = context.getClickedPos();
+    BlockState original = world.getBlockState(pos);
+
+    boolean didTransform = transform(context, original, true);
+
+    // if we made a successful transform, client can stop early
+    if (didTransform) {
+      if (world.isClientSide) {
+        return true;
+      }
+
+      // if the tool breaks or it was a campfire, we are done
+      if (ToolDamageUtil.damage(tool, 1, null, null)) {
+        return true;
+      }
+    }
+
+    // AOE transforming, run even if we did not transform the center
+    // note we consider anything effective, as hoes are not effective on all tillable blocks
+//    if (player != null && !tool.isBroken()) {
```

Comment:
Lets just hardcode to circle AOE here, using the same logic you were planning to use in fireprimer.